redfish/provider/common.go

```diff
@@ -275,12 +275,15 @@ func checkServerStatus(ctx context.Context, endpoint string, interval int, timeo
 	// Intial sleep until iDRAC reboot is triggered
 	time.Sleep(30 * time.Second)
 
-	for start := time.Now(); time.Since(start) < (time.Duration(timeout) * time.Second); {
+	for start := time.Now(); time.Since(start) < (time.Duration(timeoutInterval["timeout"]) * time.Second); {
 		tflog.Trace(ctx, "Checking server status...")
-		time.Sleep(time.Duration(interval) * time.Second)
+		time.Sleep(time.Duration(timeoutInterval["interval"]) * time.Second)
 		_, err = net.Dial("tcp", net.JoinHostPort(addr.Hostname(), addr.Scheme))
 		if err == nil {
-			return nil
+			_, err := getSystemResource(service)
+			if err == nil {
+				return nil
+			}
 		}
```

Comment:
We can break up the logic like this, rt?
```go
		if err != nil {
			continue
                }
		_, err = getSystemResource(service)
		if err == nil {
			return nil
		}
```

Reply:
yes, looks nice though